cds.test with authentication (#631)
cap-js/docs: node.js/cds-test.md

[Learn more in GET/PUT/POST.](#http-bound) {.learn-more}

#### Authenticated Endpoints
`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). This also includes the usage of [pre-definded mock users](../node.js/authentication#mock-users)

You can set the user for an authenticated request like this:

```js
await GET('/admin/Books', { auth: { username: 'alice', password: '' } })
```

> [Learn how to explicitly configure mock user in your _package.json_ file.](../node.js/authentication#mocked){.learn-more}

Reviewer: ```suggestion
[Learn how to explicitly configure mock users in your _package.json_ file.](../node.js/authentication#mocked){.learn-more}
```

[Learn more in GET/PUT/POST.](#http-bound) {.learn-more}

#### Authenticated Endpoints

> `cds.test()` uses [mocked authentication](https://pages.github.tools.sap/cap/docs/node.js/authentication#mocked). To configure authentication, you can utilize `cds.requires.auth` in the `package.json` file or directly in your test file like so:

Reviewer 2: It's a bit misleading as `cds.test` itself doesn't change/set the auth mode.  So rather:

```suggestion
`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the `package.json` file or directly in your test file like so:
```

> `cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the `package.json` file or directly in your test file like so:

Reviewer: ```suggestion
`cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file or directly in your test file as follows:
```

> ```js

Reviewer 3: I think the recommended way is to do that in package.json:

```jsonc
{
  "requires": { 
    "auth": { 
      "[development]": {
        "users":  { 
          "foo": {
            "password": "bar",
            "roles": { "admin": true, "cds.Subscriber": true },
            "attr": { "name": "Foo" }
} } } } } }
```

Author: @johannes-vogel So we should remove the other option or make it clear that the package.json is the recommended option?

Reviewer 3: I'd leave out the code variant. That's error prone and might have side effects.

> cds.requires.auth.users.foo = {

Reviewer 2: Should be `cds.env`:

```suggestion
cds.env.requires.auth.users.foo = {
```

#### Authenticated Endpoints

> `cds.test()` uses the standard authentication strategy in development mode, which is the [mocked authentication](../node.js/authentication#mocked). To configure it, use `cds.requires.auth` in the _package.json_ file as described in the documenation for the mocked authentication.

Reviewer: If mocked auth is used, can the pre-configured mock users, like alice, be used? If so, we should adapt the snippet in line 99.

Reviewer 3: yes they can be used
